python/lsst/pipe/tasks/deblendCoaddSourcesPipeline.py

```diff
@@ -161,9 +175,20 @@ def __init__(self, *, config=None):
         del self.fluxCatalogs
         del self.templateCatalogs
 
+        if config:
+            if config.useCellCoadds:
+                del self.coadds
+            else:
+                del self.coadds_cell
+                del self.backgrounds
```

Comment:
Given that you also need to dispatch on which of these connections to use in `runQuantum`, it may be better to use a single connection and let the config option change the storage class, via something like:
```
self.coadds = dataclasses.replace(self.coadds, storageClass="MutipleCellCoadd")
```
That does mean that you'd have to set the connection name in addition to setting `useCellCoadds=True` to when configuring a pipeline to use cell-based coadds, but I think it'd simplify `runQuantum` implementations and reduce the number of connections we'll have to deprecate later.

Reply:
This was the approach we had in an earlier iteration (during a live session) before switching to having the current setup of having two connections. It does make switching slightly less trivial considering the proliferating number of pipelines for LSST*Cam*

Reply:
After some further thoughts, I'm afraid I have to push back on this suggestion. Doing this would defeat the idea of having the transition seamless with a simple toggle and more over will leave the door wide open for misconfiguration when one pipeline imports from another. Excluding `assembleCellCoadd` by hand introduces some friction to the switching, but that's something I am willing to tolerate.